Automattic/wp-calypso
# Experiment: Try JPO against Jetpack site's REST API with CORS requests

This PR is a proof of concept that demonstrates how we can use Jetpack's REST API without using .com as a proxy for the purposes of JPO. Relies on CORS and JP onboarding tokens, to be used together with https://github.com/Automattic/jetpack/pull/8236.

Note: this PR doesn't follow a very good code style, and uses a lot of bad practices, but we're not planning to merge it - it's to be used only for demonstration purposes.

To test:
* Checkout this branch locally.
* Setup a WP.com site with the latest dev Jetpack on it.
* Checkout https://github.com/Automattic/jetpack/pull/8236 on your Jetpack.
* Login to your .org site.
* Go to http://calypso.localhost:3000/onboarding?url=`siteSlug`, where `siteSlug` is a slugified version of your site (for example `https://wordpress.com` becomes `wordpress.com` and `https://wordpress.com/example` becomes `wordpress.com::example`.
* Follow through as the flow will redirect you to wp-admin, generate and obtain a token, redirect to Calypso and store it, and load a "Site Title" step.
* Insert a site title in the site title field.
* Click the Save button.
* Verify the site title of your site is updated.
